handlers/search.go

```diff
@@ -127,195 +126,149 @@ func SearchAhead(w http.ResponseWriter, r *http.Request) {
-				http.Error(w, "Internal server error", http.StatusInternalServerError)
-				return
-			}
-			var tx *types.Eth1TransactionIndexed
-			tx, err = db.BigtableClient.GetIndexedEth1Transaction(txHash)
-			if err == nil && tx != nil {
-				result = &types.SearchAheadTransactionsResult{{TxHash: fmt.Sprintf("%x", tx.Hash)}}
-			}
+		var txHash []byte
+		txHash, err = hex.DecodeString(search)
+		if err != nil {
+			err = fmt.Errorf("error parsing txHash %v: %v", search, err)
+			break
+		}
+		var tx *types.Eth1TransactionIndexed
+		tx, err = db.BigtableClient.GetIndexedEth1Transaction(txHash)
+		if err != nil || tx == nil {
+			break
 		}
+		result = &types.SearchAheadTransactionsResult{{TxHash: fmt.Sprintf("%x", tx.Hash)}}
 	case "epochs":
+		epochNumber, parseErr := strconv.ParseUint(search, 10, 32)
+		if parseErr != nil {
+			break
+		}
 		result = &types.SearchAheadEpochsResult{}
-		err = db.ReaderDb.Select(result, "SELECT epoch FROM epochs WHERE CAST(epoch AS text) LIKE $1 ORDER BY epoch LIMIT 10", search+"%")
+		err = db.ReaderDb.Select(result, "SELECT epoch FROM epochs WHERE epoch = $1 ORDER BY epoch LIMIT 10", epochNumber)
 	case "validators":
 		// find all validators that have a index, publickey or name like the search-query
 		result = &types.SearchAheadValidatorsResult{}
-		indexNumeric, errParse := strconv.ParseInt(search, 10, 32)
-		if errParse == nil { // search the validator by its index
+		indexNumeric, parseErr := strconv.ParseInt(search, 10, 32)
+		if parseErr == nil { // search the validator by its index
 			err = db.ReaderDb.Select(result, `SELECT validatorindex AS index, pubkeyhex as pubkey FROM validators WHERE validatorindex = $1`, indexNumeric)
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
 		} else if thresholdHexLikeRE.MatchString(search) {
 			err = db.ReaderDb.Select(result, `SELECT validatorindex AS index, pubkeyhex as pubkey FROM validators WHERE pubkeyhex LIKE LOWER($1 || '%')`, search)
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
 		} else {
 			err = db.ReaderDb.Select(result, `
 			SELECT validatorindex AS index, pubkeyhex AS pubkey
 			FROM validators
 			LEFT JOIN validator_names ON validators.pubkey = validator_names.publickey
 			WHERE LOWER(validator_names.name) LIKE LOWER($1)
 			ORDER BY index LIMIT 10`, search+"%")
-			if err != nil {
-				logger.Errorf("error reading result data: %v", err)
-				http.Error(w, "Internal server error", http.StatusServiceUnavailable)
-				return
-			}
 		}
 	case "eth1_addresses":
-
 		var ensData *types.EnsDomainResponse
 		if utils.IsValidEnsDomain(search) {
 			ensData, _ = GetEnsDomain(search)
 			if len(ensData.Address) > 0 {
-				search = strings.Replace(ensData.Address, "0x", "", -1)
+				search = ensData.Address
```

Comment:
This change breaks the ENS logic as the search modified by the content in `ensData` does not pass the RegEx below:

Prater supports ENS and one ENS name I use for testing is `clarissa.eth`. Use this in the search and https://github.com/gobitfly/eth2-beaconchain-explorer/blob/7f866c30817481a4ee5689ca1a949c9f50e9429d/handlers/search.go#L177-L184 will cause `search` to be set to `"0x000002b4A2e3459331b1eB6066A8FdB39c7fdC5D"`. This does not pass `searchLikeRE` though.

The same applies here: https://github.com/gobitfly/eth2-beaconchain-explorer/blob/7f866c30817481a4ee5689ca1a949c9f50e9429d/handlers/search.go#L242

On Prater production searching for `clarissa.eth` shows 3 results while your branch now only shows 1. Please compare.

Reply:
fixed, ptal